### PR TITLE
Disable IPFS Pin API

### PR DESCRIPTION
**Pull request by jkpit**
Disabling pin api due to ipfs performance issues with larger object Storage.
Propable fix for #3156 while ipfs has not solved this issue (go-ipfs issue [#3874](https://github.com/ipfs/go-ipfs/issues/3874))

### CODE REVIEW

**Comment by l2isbad** [comment]
@jkpit what about  not to disable them but make optional and off by default?

**Comment by jkpit** [comment]
@l2isbad good idea, i disabled it by default

File: conf.d/python.d/ipfs.conf [hunk 1]
```diff
@@ -39,6 +39,10 @@
 # This feature is disabled by default.
 # autodetection_retry: 0
 
+# Disable/Enable IPFS pin api
+# Currently defaults to disabled due to IPFS Bug resulting in very high CPU Usage
+# pinapi: 0
```
Comment (l2isbad) [inline conf.d/python.d/ipfs.conf]: make it `bool`

File: python.d/ipfs.chart.py [hunk 2]
```diff
@@ -11,6 +11,7 @@
 # update_every = 2
 priority = 60000
 retries = 60
+pinapi = 0
```
Comment (l2isbad) [inline python.d/ipfs.chart.py]: move it to the `__init__` (like `self.do_pinapi = self.configuration.get('pinapi')`)

File: conf.d/python.d/ipfs.conf [hunk 3]
```diff
@@ -39,6 +39,10 @@
 # This feature is disabled by default.
 # autodetection_retry: 0
 
+# Disable/Enable IPFS pin api
+# Currently defaults to disabled due to IPFS Bug resulting in very high CPU Usage
```
Comment (l2isbad) [inline conf.d/python.d/ipfs.conf]: please add the link https://github.com/ipfs/go-ipfs/issues/3874

File: python.d/ipfs.chart.py [hunk 4]
```diff
@@ -11,6 +11,7 @@
 # update_every = 2
 priority = 60000
 retries = 60
+pinapi = False
```
Comment (l2isbad) [inline python.d/ipfs.chart.py]: not needed anymore

File: python.d/ipfs.chart.py [hunk 5]
```diff
@@ -110,9 +112,12 @@ def _get_data(self):
                 [('peers', 'Peers', len)],
             '/api/v0/stats/repo':
                 [('size', 'RepoSize', int), ('objects', 'NumObjects', int), ('avail', 'StorageMax', self._storagemax)],
-            '/api/v0/pin/ls':
-                [('pinned', 'Keys', len), ('recursive_pins', 'Keys', self._recursive_pins)]
         }
+        if pinapi:
```
Comment (l2isbad) [inline python.d/ipfs.chart.py]: use `self.do_pinapi`

File: conf.d/python.d/ipfs.conf [hunk 6]
```diff
@@ -39,6 +39,10 @@
 # This feature is disabled by default.
 # autodetection_retry: 0
 
+# Disable/Enable IPFS pin api
+# Currently defaults to disabled due to IPFS Bug(https://github.com/ipfs/go-ipfs/issues/3874) resulting in very high CPU Usage
+# pinapi: False
+
```
Comment (l2isbad) [inline conf.d/python.d/ipfs.conf]: Move it to the `Additionally to the above, ipfs also supports the following` section.

**Comment by l2isbad** [comment]
gj @jkpit  👍

**Comment by jkpit** [comment]
glad to help and thanks for the support @l2isbad

**Comment by ktsaou** [comment]
merge?

**Comment by ktsaou** [comment]
merged!

**Pull request merged**